src/VSIX/NpgsqlConnectionUIControl.cs

```diff
@@ -42,33 +42,41 @@ void SetProperty(object sender, EventArgs e)
             if (_loading)
                 return;  // TODO: Match with sample?
 
-            if (sender == hostTextBox)
-                Site["Host"] = hostTextBox.Text.Trim();
-            else if (sender == portNumericUpDown)
-                Site["Port"] = portNumericUpDown.Value;
-            else if (sender == databaseTextBox)
-                Site["Database"] = databaseTextBox.Text;
-            else if (sender == usernameTextBox)
-                Site["Username"] = usernameTextBox.Text;
-            else if (sender == passwordTextBox)
-                Site["Password"] = passwordTextBox.Text;
-            else if (sender == savePasswordCheckBox)
-                Site["Persist Security Info"] = savePasswordCheckBox.Checked;
-            else if (sender == windowsAuthCheckbox)
+            try
             {
-                var integratedSecurity = windowsAuthCheckbox.Checked;
-                Site["Integrated Security"] = integratedSecurity;
-                usernameTextBox.Enabled = !integratedSecurity;
-                passwordTextBox.Enabled = !integratedSecurity;
-                if (integratedSecurity)
+                if (sender == hostTextBox)
+                    Site["Host"] = hostTextBox.Text.Trim();
+                else if (sender == portNumericUpDown)
+                    Site["Port"] = portNumericUpDown.Value;
+                else if (sender == databaseTextBox)
+                    Site["Database"] = databaseTextBox.Text;
+                else if (sender == usernameTextBox)
+                    Site["Username"] = usernameTextBox.Text;
+                else if (sender == passwordTextBox)
+                    Site["Password"] = passwordTextBox.Text;
+                else if (sender == savePasswordCheckBox)
+                    Site["Persist Security Info"] = savePasswordCheckBox.Checked;
+                else if (sender == windowsAuthCheckbox)
                 {
-                    usernameTextBox.Text = "";
-                    passwordTextBox.Text = "";
+                    var integratedSecurity = windowsAuthCheckbox.Checked;
+                    Site["Integrated Security"] = integratedSecurity;
+                    usernameTextBox.Enabled = !integratedSecurity;
+                    passwordTextBox.Enabled = !integratedSecurity;
+                    if (integratedSecurity)
+                    {
+                        usernameTextBox.Text = "";
+                        passwordTextBox.Text = "";
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex.ToString());
```

Comment:
This looks like debugging stuff that should be removed before merging.

Reply:
Can you please remove this entirely? It's better to not leave this kind of thing in code, even in #ef

Reply:
I am afraid, I shouldn't.

There are times, that the connection dialog crushes just by changing field. This try block, it's not for debug reasons, it's to fix this issue.

Reply:
If that's the case, then it shouldn't be in `#if DEBUG`, no?

In any case, it would be good to understand where the actual problem comes from (what's the full exception details and stack trace?). Is it connected in any way to this PR (which changes the supported VS versions), can it be dealt with as a separate issue?

Reply:
Yes, let's do it on separate issue. I think the code and the connection dialog takes more improvement. 

I have restored the original version.